### PR TITLE
Adjusts to Microfrontend.

### PR DESCRIPTION
**Pull request by mshima**
- Add generator tests.
- Change microservice/microfrontend web server proxy to use gateway port.
- Throw error when generating Microfrontend with others client frameworks.

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/client/index.js [hunk 1]
```diff
@@ -197,6 +197,12 @@ module.exports = class JHipsterClientGenerator extends BaseBlueprintGenerator {
         this.loadTranslationConfig();
       },
 
+      checkMicrofrontend() {
+        if (this.microfrontend && !this.clientFrameworkAngular) {
```
Comment (MathieuAA) [inline generators/client/index.js:201]: if this is a new option, then this should be added to the JDL app validation

Comment (mshima) [reply generators/client/index.js:201]: In my opinion Microfrontend is too experimental to be an official jdl option.

Comment (MathieuAA) [reply generators/client/index.js:201]: alright then

File: generators/server/templates/src/main/resources/static/microservices_index.html.ejs [hunk 2]
```diff
@@ -77,7 +77,12 @@
   <p>This application is a microservice, which has been generated using <a href="https://www.jhipster.tech/" rel="noopener noreferrer">JHipster</a>.</p>
 
   <ul>
+<%_ if (microfrontend) { %>
+    <li>Microfrontends packages doesn't provide a stand-alone front-end. Packaged microfrontend should be accessed through a JHipster gateway.</li>
```
Comment (MathieuAA) [inline generators/server/templates/src/main/resources/static/microservices_index.html.ejs:81]:
```suggestion
    <li>Microfrontends packages don't provide a stand-alone front-end. Packaged microfrontend should be accessed through a JHipster gateway.</li>
```

**Review by MathieuAA — COMMENTED** [review]
found this typo

**Review by MathieuAA — APPROVED** [review]

**Pull request merged**